From 17dd5767c43059db7671a2f8121f05d90ac4f253 Mon Sep 17 00:00:00 2001
From: Olli Vanhoja <olli.vanhoja@gmail.com>
Date: Wed, 20 Oct 2021 15:47:06 +0200
Subject: [PATCH 2/2] Add ReplyWithBinaryBuffer() for sending bin bufs

---
 src/module.c      | 15 +++++++++++++++
 src/redismodule.h |  2 ++
 2 files changed, 17 insertions(+)

diff --git a/src/module.c b/src/module.c
index bda6d0a0d..22a6817fa 100644
--- a/src/module.c
+++ b/src/module.c
@@ -1553,6 +1553,20 @@ int RM_ReplyWithVerbatimString(RedisModuleCtx *ctx, const char *buf, size_t len)
     return REDISMODULE_OK;
 }
 
+/* Reply with a binary buffer than can be parsed with data-record.
+ *
+ * The function always returns REDISMODULE_OK. */
+int RM_ReplyWithBinaryBuffer(RedisModuleCtx *ctx, const char *buf, size_t len) {
+    void addReplyLongLongWithPrefix(client *c, long long ll, char prefix);
+
+    client *c = moduleGetReplyClient(ctx);
+    if (c == NULL) return REDISMODULE_OK;
+    addReplyLongLongWithPrefix(c,len,'@');
+    addReplyProto(c,buf,len);
+    addReply(c,shared.crlf);
+    return REDISMODULE_OK;
+}
+
 /* Reply to the client with a NULL.
  *
  * The function always returns REDISMODULE_OK. */
@@ -8110,6 +8124,7 @@ void moduleRegisterCoreAPI(void) {
     REGISTER_API(ReplyWithEmptyString);
     REGISTER_API(ReplyWithVerbatimString);
     REGISTER_API(ReplyWithStringBuffer);
+    REGISTER_API(ReplyWithBinaryBuffer);
     REGISTER_API(ReplyWithCString);
     REGISTER_API(ReplyWithNull);
     REGISTER_API(ReplyWithCallReply);
diff --git a/src/redismodule.h b/src/redismodule.h
index 48833553d..56e2d3dba 100644
--- a/src/redismodule.h
+++ b/src/redismodule.h
@@ -564,6 +564,7 @@ REDISMODULE_API int (*RedisModule_ReplyWithCString)(RedisModuleCtx *ctx, const c
 REDISMODULE_API int (*RedisModule_ReplyWithString)(RedisModuleCtx *ctx, RedisModuleString *str) REDISMODULE_ATTR;
 REDISMODULE_API int (*RedisModule_ReplyWithEmptyString)(RedisModuleCtx *ctx) REDISMODULE_ATTR;
 REDISMODULE_API int (*RedisModule_ReplyWithVerbatimString)(RedisModuleCtx *ctx, const char *buf, size_t len) REDISMODULE_ATTR;
+REDISMODULE_API int (*RedisModule_ReplyWithBinaryBuffer)(RedisModuleCtx *ctx, const char *buf, size_t len) REDISMODULE_ATTR;
 REDISMODULE_API int (*RedisModule_ReplyWithNull)(RedisModuleCtx *ctx) REDISMODULE_ATTR;
 REDISMODULE_API int (*RedisModule_ReplyWithDouble)(RedisModuleCtx *ctx, double d) REDISMODULE_ATTR;
 REDISMODULE_API int (*RedisModule_ReplyWithLongDouble)(RedisModuleCtx *ctx, long double d) REDISMODULE_ATTR;
@@ -792,6 +793,7 @@ static int RedisModule_Init(RedisModuleCtx *ctx, const char *name, int ver, int
     REDISMODULE_GET_API(ReplyWithString);
     REDISMODULE_GET_API(ReplyWithEmptyString);
     REDISMODULE_GET_API(ReplyWithVerbatimString);
+    REDISMODULE_GET_API(ReplyWithBinaryBuffer);
     REDISMODULE_GET_API(ReplyWithNull);
     REDISMODULE_GET_API(ReplyWithCallReply);
     REDISMODULE_GET_API(ReplyWithDouble);
-- 
2.30.1

